src/main/java/net/neoforged/accesstransformer/parser/AccessTransformerList.java

```diff
@@ -4,39 +4,57 @@
 
 import net.neoforged.accesstransformer.*;
 import org.antlr.v4.runtime.*;
-import org.apache.logging.log4j.*;
 import org.objectweb.asm.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
 
 import java.io.*;
 import java.net.*;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.charset.CodingErrorAction;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.*;
 import java.util.*;
 import java.util.function.*;
 import java.util.stream.*;
 
 public class AccessTransformerList {
-    private static final Logger LOGGER = LogManager.getLogger("AXFORM");
-    private static final Marker AXFORM_MARKER = MarkerManager.getMarker("AXFORM");
+    private static final Logger LOGGER = LoggerFactory.getLogger("AXFORM");
+    private static final Marker AXFORM_MARKER = MarkerFactory.getMarker("AXFORM");
     private final Map<Target<?>, AccessTransformer> accessTransformers = new HashMap<>();
     private Set<Type> targetedClassCache = Collections.emptySet();
 
     public void loadFromResource(final String resourceName) throws URISyntaxException, IOException {
         final Path path = Paths.get(getClass().getClassLoader().getResource(resourceName).toURI());
-        loadFromPath(path, resourceName);
+        loadFromPath(path);
     }
 
-    public void loadFromPath(final Path path, final String resourceName) throws IOException {
-        LOGGER.debug(AXFORM_MARKER,"Loading access transformer {} from path {}", resourceName, path);
-        final CharStream stream = CharStreams.fromPath(path);
+    public void loadFromPath(final Path path) throws IOException {
+        long size = Files.size(path);
+        try (ReadableByteChannel channel = Files.newByteChannel(path)) {
+            loadAT(CharStreams.fromChannel(
+                    channel,
+                    StandardCharsets.UTF_8,
+                    4096, // CharStreams.DEFAULT_BUFFER_SIZE
+                    CodingErrorAction.REPLACE,
+                    path.getFileName().toString(),
+                    size));
```

Comment:
Where is that coming from? :P

Reply:
`CharStreams.fromPath`, which unfortunately uses `path.toString()` (i.e. the full path) as the source name.